### ISSUE TITLE
bug(bridge): Doctrine::update()

### ISSUE THREAD

**Issue by jvancoillie** [Guikingone/SchedulerBundle#45]
Hi, 

Task update fails due to lock mode ([getWriteLockSQL()](https://github.com/Guikingone/SchedulerBundle/blob/35aacb893d5bc1fade7ea2faac07a0223ddff1ec/src/Bridge/Doctrine/Transport/Connection.php#L291)) which I think is only available on a SQL SELECT command.

that throws the following exception 
```
SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'FOR UPDATE' at line 1
```

**Comment by Guikingone** [comment]
Hi @jvancoillie 👋🏻 

Hum, strange behaviour here 🤔 

This method is tested on [SQLite](https://github.com/Guikingone/SchedulerBundle/blob/main/tests/Bridge/Doctrine/Transport/ConnectionIntegrationTest.php#L144) and seems to work, do you have more informations about the stack used? Maybe the SGBQ driver and so on? 

Thanks and sorry for the issue 🙂

**Comment by jvancoillie** [comment]
no problem, I discover the bundle and try to understand how it works so that I can use it on a current project.

It is possible that the problem does not come from the bundle but from the use that I make of it.

Currently I am on version 5.2.6 of symfony with php 7.4.15 and a mariadb in version 10.5.8 (docker mariadb:10.5.8) with the following overrided configuration:
```
[mysqld]
sql-mode = "STRICT_TRANS_TABLES, NO_ZERO_IN_DATE, ERROR_FOR_DIVISION_BY_ZERO, NO_ENGINE_SUBSTITUTION"
character-set-server = utf8
default-authentication-plugin = mysql_native_password
```

**Comment by jvancoillie** [comment]
I tested without the writeLockSQL and when updating the task it does not detect any change which throws the exception "The given task cannot be updated as the identifier or the body is invalid" and rollback because the rowCount ( ) equal to 0. 
```
[2021-03-30T16:55:45.811072+00:00] doctrine.DEBUG: "START TRANSACTION" [] []
[2021-03-30T16:55:45.811819+00:00] doctrine.DEBUG: UPDATE scheduler_tasks SET body = :body WHERE task_name = :name {":name":"activation",":body":"{\"body\":{\"command\":\"app:ac [...]"} []
[2021-03-30T16:55:45.812709+00:00] doctrine.DEBUG: "ROLLBACK" [] []
```

**Comment by Guikingone** [comment]
Do you have the whole process that lead to this error? 🤔 

It is during a specific command?

**Comment by Guikingone** [comment]
Oh, that's what I call a great issue 😄 

Ok, got it, by default and if `single_run` is set to `true`, the task is unscheduled right after the execution, thing is, the process of updating the task is not aborted and the connection cannot do the update 🤔 

Looks like a "human-related" bug on my side 😄

**Comment by Guikingone** [comment]
> Désolé pour mon anglais médiocre ;)

Aucun souci pour l'anglais, le mien n'est pas meilleur (et je ne parle pas de l'accent made in "Chud de la France" 😄 ). 

The `single_run` is made to execute the task once and only once, BTW, you create an interesting use case were we can have tasks that must be executed only once when deploying / booting the scheduler / etc. 

Could be interesting to thing about it for next releases 🙂

**Comment by jvancoillie** [comment]
Great, what is a shame is that I will no longer have information on the single_run command. Currently, when I list the scheduler I find my activation command (certainly because of the bug). 


<img width="1164" alt="Capture d’écran 2021-03-30 à 19 40 00" src="https://user-images.githubusercontent.com/21243943/113032104-c2d4aa00-918f-11eb-8111-23ed2bf2c742.png">
After the correction, I should not have any more information about this task, is that right?

it could have been interesting to be able to consult the execution time of this command and to keep track of the last execution

**Comment by Guikingone** [comment]
Hum, in fact, the behaviour can be changed / improved, I'm thinking about the fact that a `single_run` task can be disabled once executed rather than removed from the scheduler 🤔

**Comment by Guikingone** [comment]
Small update, the fix has been merged, as the core API has been changed (`unschedule` to `pause`), the fix is moved to `0.4` for release 🙂

**Comment by jvancoillie** [comment]
@Guikingone thanks
